account_test.go

```diff
@@ -3,101 +3,65 @@ package stripe
-	if err != nil {
-		t.Error(err)
-	}
-
-	if account.ID != "acct_1234" {
-		t.Errorf("Problem deserializing account, got ID %v", account.ID)
-	}
-
-	if !account.ExternalAccounts.More {
-		t.Errorf("Problem deserializing account, wrong value for list More")
-	}
-
-	if len(account.ExternalAccounts.Values) != 2 {
-		t.Errorf("Problem deserializing account, got wrong number of external accounts")
-	}
-
-	bankAccount := account.ExternalAccounts.Values[0]
-	if bankAccount == nil {
-		t.Errorf("Problem deserializing account, didn't get a bank account")
-	}
-	if "ba_1234" != bankAccount.ID {
-		t.Errorf("Problem deserializing account, got bank account ID %v", bankAccount.ID)
-	}
+	assert.NoError(t, err)
 
-	card := account.ExternalAccounts.Values[1]
-	if card == nil {
-		t.Errorf("Problem deserializing account, didn't get a card")
-	}
+	assert.Equal(t, "acct_123", account.ID)
+	assert.Equal(t, true, account.ExternalAccounts.More)
 
-	if "card_1234" != card.ID {
-		t.Errorf("Problem deserializing account, got card ID %v", card.ID)
-	}
+	assert.Equal(t, 2, len(account.ExternalAccounts.Values))
+	assert.Equal(t, "ba_123", account.ExternalAccounts.Values[0].ID)
+	assert.Equal(t, "card_123", account.ExternalAccounts.Values[1].ID)
 }
 
-func TestAddressAppendDetails(t *testing.T) {
-	actual := &RequestValues{}
-	expected := &RequestValues{}
-
-	address := &Address{}
-	address.AppendDetails(actual, "test_address")
-
-	// First test the empty case. All empty fields should be respected while
-	// encoding.
-	if expected.Encode() != actual.Encode() {
-		t.Errorf("Problem encoding address, got: %v", actual.Encode())
-	}
-
-	address = &Address{
-		Line1:   "test_line1",
-		Line2:   "test_line2",
-		City:    "test_city",
-		State:   "test_state",
-		Zip:     "test_zip",
-		Country: "test_country",
-		Town:    "test_town",
+func TestIdentityDocument_Appendto(t *testing.T) {
+	{
+		params := &IdentityDocument{ID: "file_123"}
+		body := &form.Values{}
+		params.AppendTo(body,
+			[]string{"legal_entity", "additional_owners", "0", "verification", "document"})
+		t.Logf("body = %+v", body)
+		assert.Equal(t,
+			[]string{"file_123"},
+			body.Get("legal_entity[additional_owners][0][verification][document]"),
+		)
 	}
-	address.AppendDetails(actual, "test_address")
-
-	expected.Add("test_address[line1]", "test_line1")
-	expected.Add("test_address[line2]", "test_line2")
-	expected.Add("test_address[city]", "test_city")
-	expected.Add("test_address[state]", "test_state")
-	expected.Add("test_address[postal_code]", "test_zip")
-	expected.Add("test_address[country]", "test_country")
-	expected.Add("test_address[town]", "test_town")
+}
 
-	if expected.Encode() != actual.Encode() {
-		t.Errorf("Problem encoding address, got: %v", actual.Encode())
+func TestPayoutScheduleParams_AppendTo(t *testing.T) {
+	{
+		params := &PayoutScheduleParams{MinimumDelay: true}
+		body := &form.Values{}
+		form.AppendTo(body, params)
+		t.Logf("body = %+v", body)
```

Comment:
It's not strictly necessary, but I've left a couple logging statements in because they're really handy in the case of a failure, and Go is smart enough to hide their output _unless_ there is a failure.

Reply:
oh that's a neat feature